src/client.cpp

```diff
@@ -411,16 +411,14 @@ void Client::step(float dtime)
 	/*
 		Get events
 	*/
-	for(;;) {
-		ClientEnvEvent event = m_env.getClientEvent();
-		if(event.type == CEE_NONE) {
-			break;
-		}
-		else if(event.type == CEE_PLAYER_DAMAGE) {
-			if(m_ignore_damage_timer <= 0) {
-				u8 damage = event.player_damage.amount;
+	while (m_env.hasClientEnvEvents()) {
+		ClientEnvEvent envEvent = m_env.getClientEnvEvent();
+
+		if (envEvent.type == CEE_PLAYER_DAMAGE) {
+			if (m_ignore_damage_timer <= 0) {
+				u8 damage = envEvent.player_damage.amount;
 
-				if(event.player_damage.send_to_server)
+				if (envEvent.player_damage.send_to_server)
```

Comment:
Note: renamed from event to envEvent because it shadows ClientEvent just after this code part